src/commonMain/kotlin/teksturepako/pakku/api/projects/ProjectType.kt

```diff
@@ -4,20 +4,20 @@ import teksturepako.pakku.api.data.ConfigFile
 import kotlin.reflect.KProperty1
 
 enum class ProjectType(
-    val prettyName: String, private val defaultPath: String, private val pathConfig: KProperty1<ConfigFile, String?>
+    val prettyName: String, val defaultPath: String
 )
 {
-    MOD("mod", "mods", ConfigFile::modsPath),
-    RESOURCE_PACK("resource pack", "resourcepacks", ConfigFile::resourcePacksPath),
-    DATA_PACK("data pack", "datapacks", ConfigFile::dataPacksPath),
-    WORLD("world", "saves", ConfigFile::worldsPath),
-    SHADER("shader pack", "shaderpacks", ConfigFile::shadersPath);
+    MOD("mod", "mods"),
+    RESOURCE_PACK("resource pack", "resourcepacks"),
+    DATA_PACK("data pack", "datapacks"),
+    WORLD("world", "saves"),
+    SHADER("shader pack", "shaderpacks");
 
     fun getPathString(configFile: ConfigFile?): String
     {
         return if (configFile == null) return defaultPath else
         {
-            pathConfig.get(configFile) ?: defaultPath
+            configFile.paths.getOrDefault(this, defaultPath)
```

Comment:
It should accept lowercase project type names.

Reply:
Hm? It's using project type as the key of the map. Why we need the name here

Reply:
Understand. I'll create a serializer for ProjectType.

Reply:
You don't need to. Just add `@Serializable` and `@SerialName("<project type>")`

Reply:
It should be also in plural.
```json
{
  "paths": {
      "mods": ""
  }
}
```

Reply:
Should I enable https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-builder/decode-enums-case-insensitive.html

Reply:
I would do it like this:

```kt
val paths: MutableMap<String, String> = mutableMapOf(),
```

Then add `serialName` property to `ProjectType` and use
```kt
ProjectType.valueOf()
```

Reply:
Like:

```kt
enum class ProjectType(
    val serialName: String, val prettyName: String, private val defaultPath: String
)
```

This way paths can be used for different things in the future

Reply:
Why not use the name.lowercase() directly for the `serialName`?

Reply:
> It should be also in plural.

Reply:
```
{
  "paths": {
    "mods":
    "resource_packs":
    etc.
  }
}
```